src/libtiled/tilelayer.h

```diff
@@ -568,6 +571,19 @@ inline void TileLayer::setSize(const QSize &size)
     mHeight = size.height();
 }
 
+inline QRect TileLayer::bounds() const
+{
+    QRect bound = mBounds.boundingRect();
+
+    int x = std::min(0, bound.x());
+    int y = std::min(0, bound.y());
+    int width = std::max(mWidth, bound.width());
+    int height = std::max(mHeight, bound.height());
+
+    return QRect(x, y, width, height).translated(mX, mY);
```

Comment:
I was talking about the `.translated(mX, mY)` part. Because it's there, you're doing `layer->bounds().translated(-(layer->x()), -(layer->y()))` in the renderer. You could leave out both calls to `translate` for the same result.